dev-packages/cli/src/commands/update-next.ts

```diff
@@ -0,0 +1,119 @@
+
+import * as fs from 'fs';
+import { glob } from 'glob';
+import * as jq from 'node-jq';
+import * as path from 'path';
+import * as sh from 'shelljs';
+import { baseCommand, configureShell } from '../util/command-util';
+import { LOGGER, configureLogger } from '../util/logger';
+import { validateGitDirectory } from '../util/validation-util';
+export const UpdateNextCommand = baseCommand() //
+    .name('updateNext')
+    .alias('u')
+    .description('Updates all `next` dependencies in GLSP project to the latest version')
+    .argument('[rootDir]', 'The repository root', validateGitDirectory, process.cwd())
+    .option('-v, --verbose', 'Enable verbose (debug) log output', false)
+    .action(updateNext);
+
+export async function updateNext(rootDir: string, options: { verbose: boolean }): Promise<void> {
+    configureLogger(options.verbose);
+    configureShell({ silent: false, fatal: true });
+
+    LOGGER.info('Updating next dependencies ...');
+    rootDir = path.resolve(rootDir);
+    const packages = await getWorkspacePackages(rootDir);
+    LOGGER.debug(`Scanning ${packages.length} packages to derive resolutions`, packages);
+    const resolutions = await getResolutions(packages);
+    if (!resolutions) {
+        LOGGER.info('No next dependencies found');
+        return;
+    }
+    LOGGER.info('Upgrade and rebuild packages ...');
+    const packageJson = fs.readFileSync(path.join(rootDir, 'package.json'), 'utf-8');
+    LOGGER.debug('Updating package.json with resolutions', resolutions);
+    fs.writeFileSync(path.join(rootDir, 'package.json'), JSON.stringify({ ...JSON.parse(packageJson), resolutions }, undefined, 2));
+    LOGGER.debug('Running yarn install');
+    sh.exec('yarn install --ignore-scripts');
+    LOGGER.debug('Reverting package.json');
+    sh.exec('git checkout HEAD -- package.json');
+    LOGGER.debug('Rebuild to update yarn.lock');
+    sh.exec('yarn');
+    LOGGER.info('Upgrade successfully completed');
+}
+
+async function getWorkspacePackages(rootDir: string): Promise<string[]> {
+    const rootPackage = path.join(rootDir, 'package.json');
+    const packages = [rootPackage];
+    if (!fs.existsSync(rootPackage)) {
+        LOGGER.error('No package.json found in root directory');
+        process.exit(1);
+    }
+    const workspaces = await getWorkspaceConfig(rootPackage);
+    if (workspaces) {
+        workspaces
+            .map(workspace => `${workspace}/**/package.json`)
+            .forEach(pattern => {
+                glob.sync(pattern, {
+                    cwd: rootDir,
+                    ignore: ['**/node_modules/**']
+                }).forEach(packageJson => packages.push(path.join(rootDir, packageJson)));
+            });
+    }
+
+    return [...new Set(packages)];
+}
+
+async function getResolutions(packages: string[]): Promise<Record<string, string>> {
+    let dependencies: string[] = [];
+    for (const pkg of packages) {
+        const deps = await jq.run(
```

Comment:
This will fail if you have a `package.json` that does not contain any dependencies, for example `glsp-server-node/examples/workflow-server-bundled/package.json`:

```
[...]/eclipse-glsp/glsp/node_modules/node-jq/lib/exec.js:31
                    return reject(Error(stderr));
                                  ^

Error: jq: error (at [...]/eclipse-glsp/glsp-server-node/examples/workflow-server-bundled/package.json:52): null (null) has no keys
    at ChildProcess.<anonymous> ([...]/eclipse-glsp/glsp/node_modules/node-jq/lib/exec.js:31:35)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

Either we can fix the query (preferred) or we need to do a try-catch around it.

Reply:
Query is fixed